ntoskrnl/se/tokencls.c

```diff
@@ -905,9 +905,16 @@ NtQueryInformationToken(
             }
 
             case TokenGroupsAndPrivileges:
-                DPRINT1("NtQueryInformationToken(TokenGroupsAndPrivileges) not implemented\n");
-                Status = STATUS_NOT_IMPLEMENTED;
+            {
+                PTOKEN_GROUPS_AND_PRIVILEGES GroupsAndPrivs = (PTOKEN_GROUPS_AND_PRIVILEGES)TokenInformation;
+
+                DPRINT("NtQueryInformationToken(TokenGroupsAndPrivileges)\n");
+                Status = SeQueryTokenGroupsAndPrivileges(Token,
```

Comment:
Implementation could be added directly in that case block (as done for all others), in case nowhere else in the kernel you need to use SeQueryTokenGroupsAndPrivileges
(Also, that function, since it would be private, internal and ROS-specific, would be "Sep...", not "Se".)